pkg/sql/sqlbase/structured.go

```diff
@@ -4095,3 +4096,13 @@ func GenerateUniqueConstraintName(prefix string, nameExistsFunc func(name string
 	}
 	return name
 }
+
+// GetLogicalColumnID returns the LogicalColumnID if it has one, otherwise
+// returns the ID. The regular ColumnID is used for backwards compatibility.
+func (desc *ColumnDescriptor) GetLogicalColumnID() ColumnID {
```

Comment:
This function will basically be created by proto already -- i think you want to to populate this when you read the descriptor from disk